### PR TITLE
CLIENTS-1505: Add broker config delete endpoint

### PR DESCRIPTION
**Pull request by agg111**
### What
Add delete endpoint for broker configs.

### Testing

1. Added unit tests for manager and resource classes.
2. Added integration tests.

### References
Jira - https://confluentinc.atlassian.net/browse/CLIENTS-1505

### CODE REVIEW

File: kafka-rest/src/test/java/io/confluent/kafkarest/controllers/BrokerConfigManagerImplTest.java [hunk 1]
```diff
@@ -37,14 +37,12 @@
 import javax.ws.rs.NotFoundException;
 import org.apache.kafka.clients.admin.Admin;
 import org.apache.kafka.clients.admin.AlterConfigOp;
-import org.apache.kafka.clients.admin.AlterConfigOp.OpType;
```
Comment (agg111) [inline kafka-rest/src/test/java/io/confluent/kafkarest/controllers/BrokerConfigManagerImplTest.java:40]: Had added this extra import in the last PR, so removed it, and replaced its usage in the code to `AlterConfigOp.OpType`

File: kafka-rest/src/test/java/io/confluent/kafkarest/controllers/BrokerConfigManagerImplTest.java [hunk 2]
```diff
@@ -37,14 +37,12 @@
...
 import org.apache.kafka.clients.admin.AlterConfigsResult;
 import org.apache.kafka.clients.admin.Config;
 import org.apache.kafka.clients.admin.ConfigEntry;
 import org.apache.kafka.clients.admin.DescribeConfigsResult;
 import org.apache.kafka.common.KafkaFuture;
 import org.apache.kafka.common.config.ConfigResource;
-import org.apache.kafka.common.config.ConfigResource.Type;
```
Comment (agg111) [inline kafka-rest/src/test/java/io/confluent/kafkarest/controllers/BrokerConfigManagerImplTest.java:47]: Same as the `AlterConfigOp.OpType`, had added an extra import, so removed it and modified its usages to `ConfigResource.Type`

File: kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/BrokerConfigsResourceIntegrationTest.java [hunk 3]
```diff
@@ -206,6 +206,121 @@ public void getBrokerConfig_nonExistingCluster_throwsNotFound() {
     Assert.assertEquals(Status.NOT_FOUND.getStatusCode(), response.getStatus());
   }
 
+  @Test
+  public void
+  getThenUpdateThenGetThenResetThenGet_existingConfig_returnsDefaultThenUpdatesThenReturnsUpdatedThenReturnsDefault
```
Comment (agg111) [inline kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/BrokerConfigsResourceIntegrationTest.java:211]: nit - Do we modify the test name to something like `getUpdateReset_withExistingConfig` or `compositeBrokerConfigRequests_withExistingConfig` and comments in the test like - ` // call to get broker configs` , ` // call to update broker configs` ...

Comment (rigelbm) [reply kafka-rest/src/test/java/io/confluent/kafkarest/integration/v3/BrokerConfigsResourceIntegrationTest.java:211]: `getUpdateReset_withExistingConfig` sounds good.

**Comment by agg111** [comment]
Thank you Rigel, will merge once build passes.

**Pull request merged**